app/mainwindow.cpp

```diff
@@ -1211,18 +1179,21 @@ void MainWindow::setupHtmlPreviewThemes()
     int key = 1;
     bool separatorAdded = false;
     foreach(const QString &themeName, themeCollection->themeNames()) {
-        if (!separatorAdded && !themeCollection->theme(themeName).isBuiltIn()) {
-            addSeparatorAfterBuiltInThemes();
-            separatorAdded = true;
-        }
-
         QAction *action = ui->menuStyles->addAction(themeName);
         action->setShortcut(QKeySequence(tr("Ctrl+%1").arg(key++)));
         action->setCheckable(true);
         action->setActionGroup(stylesGroup);
         connect(action, &QAction::triggered,
                 this, &MainWindow::themeChanged);
+
+        if (!separatorAdded && !themeCollection->theme(themeName).isBuiltIn()) {
```

Comment:
You seem to not have my bugfix commit 37ecdbf8fb5dfab09cf525626c83305aefbf7f97 in your branch. Please update your branch so that this fix does not get reverted again.